CS483/CS483/Kartaclysm/Components/ComponentDriverStatMods.cpp

```diff
@@ -0,0 +1,90 @@
+//----------------------------------------------------------------------------
+// ComponentDriverStatMods
+// Author: David MacIntosh
+//
+// Component for the stat modifiers of a driver
+//----------------------------------------------------------------------------
+
+#include "ComponentDriverStatMods.h"
+
+#include "ComponentKartController.h"
+
+namespace Kartaclysm
+{
+	ComponentDriverStatMods::ComponentDriverStatMods(
+		HeatStroke::GameObject* p_pGameObject,
+		int p_iMaxSpeed,
+		int p_iAcceleration,
+		int p_iHandling,
+		int p_iDurability)
+		:
+		Component(p_pGameObject),
+		m_iMaxSpeedStat(p_iMaxSpeed),
+		m_iAccelerationStat(p_iAcceleration),
+		m_iHandlingStat(p_iHandling),
+		m_iDurabilityStat(p_iDurability)
+	{
+		ComponentKartController* kartController = (ComponentKartController*)m_pGameObject->GetParent()->GetComponent("GOC_KartController");
+
+		kartController->SetDriverStatMods(m_iMaxSpeedStat, m_iAccelerationStat, m_iHandlingStat, m_iDurabilityStat);
```

Comment:
Any calls to other components should be done in Init(), otherwise you can't guarantee if that component has been made yet. There should also be a way to handle if it's a nullptr (assert or something). Same for KartStats.cpp.